std/path.d

```diff
@@ -2542,50 +2613,36 @@ string relativePath(CaseSensitive cs = CaseSensitive.osDefault)
         .joiner(dirSeparator)
         .byChar;
 
-    // Return (r1 ~ dirSeparator ~ r2)
-    auto result = chain(r1, dirSeparator[0 .. $ * !(r1.empty || r2.empty)].byChar, r2).array;
-    return ((r) @trusted => cast(string)r)(result);
+    // Return (r1 ~ sep ~ r2)
+    return choose(choosePath, path.byCodeUnit, chain(r1, sep.byChar, r2));
 }
 
+///
 unittest
 {
-    import std.exception;
-    assert (relativePath("foo") == "foo");
+    import std.array;
     version (Posix)
     {
-        assert (relativePath("foo", "/bar") == "foo");
-        assert (relativePath("/foo/bar", "/foo/bar") == ".");
-        assert (relativePath("/foo/bar", "/foo/baz") == "../bar");
-        assert (relativePath("/foo/bar/baz", "/foo/woo/wee") == "../../bar/baz");
-        assert (relativePath("/foo/bar/baz", "/foo/bar") == "baz");
-        assertThrown(relativePath("/foo", "bar"));
-
-        assertCTFEable!(
-        {
-            assert (relativePath("/foo/bar", "/foo/baz") == "../bar");
-        });
+        assert (toRelativePath("foo", "/bar").array == "foo");
```

Comment:
I don't like .equal for this because 1) it autodecodes and hence throws and 2) it shows how to get the string that lots of users may actually want.